hyperspy/component.py

```diff
@@ -910,20 +907,18 @@ def _component2plot(self, axes_manager, out_of_range2nans=True):
         if axes_manager is not self.model.axes_manager:
             old_axes_manager = self.model.axes_manager
             self.model.axes_manager = axes_manager
-            self.charge()
```

Comment:
This was actually a bug: `charge` was renamed to `fetch_stored_values`, but not here. Removing it causes another issue. I would suggest to simply replace `charge` with `fetch_stored_values`.